tests/playbooks/tests_states.yml

```diff
@@ -29,7 +29,7 @@
               # Device should be present because of autoconnect: true by
               # default for NM (this might be considered a bug)
               - what: tasks/assert_device_present.yml
-                when: "{{ network_provider == 'nm' }}"
```

Comment:
Is this a spacing error?

Reply:
`ansible-lint` indicates that it is a jinja[spacing] error. And the code change that I made is also suggested by `ansible-lint`.

Reply:
It's not really a spacing error - it is that a `when` (like assert `that`, `changed_when`, `failed_when`, etc.) are already evaluated by Ansible as a Jinja context - so you do not need to use `when: "{{ network_provider == 'nm' }}"` when `when: network_provider == 'nm'` will suffice

Reply:
I think the changes are fine but we should modify the commit message.

Something like:

` ansible-lint: Fix expressions already evaluated by Ansible as Jinja context`

or maybe is this too long? I think mentioning that it is a spacing error is confusing.. 

Any suggestion?

Thanks!

Reply:
How does ansible-lint report this error? If Ansible-lint calls this spacing, then it is fine by me. I would welcome to just show the message from ansible-lint here and be done if it shows this as a spacing issue.

Reply:
ansible-lint reports it as jinja[spacing].. so maybe it is fine (it seems confusing to me) but well, I suppose it makes sense because ansible-lint is reporting it

Reply:
Can you please paste the line how it is reporting this?

Reply:
It seems these are actually warnings not failures FWIW. I updated the commits and summary.

Reply:
```
jinja[spacing]: Jinja2 spacing could be improved: {{ lsr_assert_when|default([]) }} -> {{ lsr_assert_when | default([]) }} (warning)
tests/tasks/run_test.yml:41 Jinja2 template rewrite recommendation: `{{ lsr_assert_when | default([]) }}`.
```

Reply:
The error message is:
```
jinja[spacing]: Jinja2 spacing could be improved: {{ network_provider == 'nm' }} -> network_provider == 'nm' (warning)
tests/playbooks/tests_states.yml:31 Task/Handler: include_tasks tasks/run_test.yml
```